common/data_refinery_common/models/models.py

```diff
@@ -719,6 +719,17 @@ def save(self, *args, **kwargs):
         self.last_modified = current_time
         return super(OriginalFile, self).save(*args, **kwargs)
 
+    def set_downloaded(self, absolute_file_path, filename = None):
+        """ Marks the file as downloaded, if `filename` is not provided it will
+        be parsed from the `absolute_file_path` """
+        self.is_downloaded = True
+        self.is_archive = False
```

Comment:
Woah, this seems dangerous because there's nothing about the function name or docstring that indicates it's not appropriate to call this on an archive file.